include/openvswitch/ovs-p4rt.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Defines the public interface to an externally-supplied module
+ * that permits OvS to communicate with the P4 control plane.
+ */
+
+#ifndef OPENVSWITCH_OVS_P4RT_H
+#define OPENVSWITCH_OVS_P4RT_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+struct p4_ipaddr {
+  uint8_t family;
+  uint8_t prefix_len;
+  uint32_t v4addr;
+};
+
+struct tunnel_info {
+    uint32_t ifindex;
+    uint32_t port_id;
+    struct p4_ipaddr local_ip;
+    struct p4_ipaddr remote_ip;
+    uint16_t dst_port;
+    uint16_t vni;
+};
+
+struct vlan_info {
+    uint32_t vlan_id;
+};
+
+struct mac_learning_info {
+    bool is_tunnel;
+    bool is_vlan;
+    uint8_t mac_addr[6];
+    union {
+        struct tunnel_info tnl_info;
+        struct vlan_info vln_info;
+    };
+};
+
+// Function declarations
+extern void ConfigFdbTableEntry(struct mac_learning_info learn_info,
```

Comment:
two thoughts:
1. The ovs community might prefer snake-case to CamelCase for the function names.
2. I'm surprised by the `void` return type. This means the functions can't report failure to the caller.

Reply:
About snake_case, I will change it as part of separate PR after this drop. Do you think only exposed function should be snake_case or whole sidecar library should be to be consistent with exposed functions?

Thinking about return type, OVS code can't do much if P4 programming of table entry failed. Once logging is enabled, we will print log in ovs-p4rt.log in case of failure. Can you think of any other approach here in case of failure?